packages/core/src/utils/git.js

```diff
@@ -0,0 +1,26 @@
+const logger = require('./logger');
+
+/**
+ * Wrapper around simple-git providing helper functions to retrieve repo remote URLs.
+ */
+module.exports = {
+  /**
+   * Returns the contents of a remote file, undefined otherwise.
+   * See: https://git-scm.com/docs/git-cat-file for accepted values for each input.
+   */
+  async getRemoteBranchFile(simpleGit, type, remote, branch, fileName) {
+    const catFileTarget = `${remote}/${branch}:${fileName}`;
+    return simpleGit.catFile([type, catFileTarget])
```

Comment:
Not too familiar with `CNAME` file and `cat-file`, give me some time to take a look at that!
Perhaps one of the other maintainers @acjh @marvinchin might have across it before though

Reply:
This isn't something I'm familiar with either, unfortunately :(